src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Debug/KeyQuotedValuePairListEncoding.cs

```diff
@@ -0,0 +1,51 @@
+﻿// Licensed to the .NET Foundation under one or more agreements. The .NET Foundation licenses this file to you under the MIT license. See the LICENSE.md file in the project root for more information.
+
+using System.Text.RegularExpressions;
+
+namespace Microsoft.VisualStudio.ProjectSystem.Debug;
+
+/// <summary>
+///  Encoding corresponding a format of keys eith a respective value that
+///  is surrounded in quotes, ex. key1="value1"
+/// </summary>
+internal sealed class KeyQuotedValuePairListEncoding
+{
+    public static KeyQuotedValuePairListEncoding Instance { get; } = new();
+
+    public IEnumerable<(string Name, string Value)> Parse(string input)
+    {
+        if (string.IsNullOrWhiteSpace(input))
+        {
+            yield break;
+        }
+
+        input = input.Replace("\"", "") + ",";
+        var regex = new Regex(@"(\S+?\s*)\=(\S+?\s*),");
```

Comment:
This supports empty names and values. Assuming that's intentional, the regular expression can be simplified a little:

```suggestion
        var regex = new Regex(@"(\S*\s*)\=(\S*\s*),");
```

This is because `+` matches one or more, and `?` makes that optional. In combination, they are the same as `*` which match zero or more.

---

The use of `(\S*\s*)` means that trailing white space is captured, but not leading white space. Is that intentional?

---

I'm concerned that this won't work when the name or value has either the `=`, `,` or `"` characters within.

---

Named capture groups can make regular expressions easier to understand:

```suggestion
        var regex = new Regex(@"(?<name>\S*\s*)\=(<value>\S*\s*),");
```

Then later you write:

```c#
            string name = match.Groups["name"].Value;
            string value = match.Groups["value"].Value;
```

---

Note that I replaced "key" with "name" on the assumption that it's valid for a user to have duplicate names in the data. Doing so would not be invalid within the project file, and we should respect that.